### PR TITLE
Limit SDK/platform tests to the SDK HEAD

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Hopefully the comment in the code puts it best.

```
# Instead of testing the full cartesian product of all SDK versions with
# all platform (~= Sandbox/JSON API) versions, we test the latest version of
# each with all versions of the other. This gives us a reasonable feedback
# with regards to maintaining backwards-compatibility without causing the
# test runs to grow quadratically.
```

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compatibility/BUILD [hunk 1]
```diff
@@ -52,12 +52,18 @@ config_setting(
...
+# the latest one. This gives us a reasonable feedback with regards to
+# maintaining backwards-compatibility without causing the test runs to grow
+# exponentially.
+sdk_head = "0.0.0"
+
 [
     sdk_platform_test(
         platform_version = platform_version,
-        sdk_version = sdk_version,
+        sdk_version = sdk_head,
```
Comment (cocreature) [inline compatibility/BUILD:65]: Don’t we want to test in both directions? sdk head vs all platforms and platform head vs all sdks.

File: compatibility/BUILD [hunk 2]
```diff
@@ -52,12 +52,18 @@ config_setting(
     for sdk_version in sdk_versions
 ]
 
+# Instead of testing the full cartesian product of all SDK versions with
+# all platform (~= Sandbox/JSON API) versions, we pin the SDK version to be
+# the latest one. This gives us a reasonable feedback with regards to
+# maintaining backwards-compatibility without causing the test runs to grow
```
Comment (cocreature) [inline compatibility/BUILD:58]: they grow quadratically not exponentially :slightly_smiling_face:

File: compatibility/BUILD (hunk 1 above)
Comment (stefanobaghino-da) [reply compatibility/BUILD:65]: My bad, thanks.

File: compatibility/BUILD (hunk 2 above)
Comment (stefanobaghino-da) [reply compatibility/BUILD:58]: Of course, thanks.

**Review by cocreature — APPROVED** [review]
Thank you!

File: compatibility/BUILD (hunk 1 above)
Comment (stefanobaghino-da) [reply compatibility/BUILD]: Push forced and fixed in https://github.com/digital-asset/daml/pull/8893/commits/599a135d35318d22e810e6bc86026621b5746056

I guess there could be a more elegant way of doing the list comprehension in a single expression, but I thought of keeping it simple. Let me know if you prefer it to be a single expression.

File: compatibility/BUILD (hunk 2 above)
Comment (stefanobaghino-da) [reply compatibility/BUILD]: Push forced and fixed in https://github.com/digital-asset/daml/pull/8893/commits/599a135d35318d22e810e6bc86026621b5746056

I changed the PR description to match the commit message.

File: compatibility/BUILD [hunk 3]
```diff
@@ -52,13 +52,27 @@ config_setting(
...
 [
     sdk_platform_test(
         platform_version = platform_version,
+        sdk_version = head,
+    )
+    for platform_version in platform_versions
+]
+
+[
+    sdk_platform_test(
```
Comment (cocreature) [inline compatibility/BUILD:71]: you now have 0.0.0 against 0.0.0 twice which fails with
```
ERROR: Traceback (most recent call last):
	File "/Users/vsts/agent/_work/1/s/compatibility/BUILD", line 71, column 22, in <toplevel>
		sdk_platform_test(
	File "/Users/vsts/agent/_work/1/s/compatibility/bazel_tools/testing.bzl", line 415, column 23, in sdk_platform_test
		client_server_test(
	File "/private/var/tmp/_bazel_vsts/6938b4f491cf5cda3150ff93fe52f44b/external/daml/bazel_tools/client_server/client_server_test.bzl", line 57, column 19, in client_server_test
		sh_inline_test(
	File "/private/var/tmp/_bazel_vsts/6938b4f491cf5cda3150ff93fe52f44b/external/daml/bazel_tools/sh/sh.bzl", line 54, column 22, in sh_inline_test
		_sh_inline_script(
Error in _sh_inline_script: _sh_inline_script rule 'ledger-api-test-tool-0.0.0-platform-0.0.0_script' in package '' conflicts with existing _sh_inline_script rule, defined at /Users/vsts/agent/_work/1/s/compatibility/BUILD:63:22
ERROR: package contains errors: 
ERROR: error loading package '': Package '' contains errors
```

Comment (stefanobaghino-da) [reply compatibility/BUILD:71]: Yup, on it, thanks.

Comment (stefanobaghino-da) [reply compatibility/BUILD:71]: I guess it does not make sense to test the latest platform and SDK together, that's not a compatibility test.

Comment (cocreature) [reply compatibility/BUILD:71]: I think it’s still a useful sanity check to make sure that the tests work as they should.

Comment (stefanobaghino-da) [reply compatibility/BUILD:77]: Addressed by https://github.com/digital-asset/daml/pull/8893/commits/c18bf26c0fe7c603e3f2efe407c89d5b39af968f

**Pull request merged**